src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -187,28 +187,33 @@ private BoundExpression ToBadExpression(BoundExpression expr, LookupResultKind r
             Debug.Assert(resultKind != LookupResultKind.Viable);
 
             TypeSymbol resultType = expr.Type;
-            BoundKind exprKind = expr.Kind;
+            BoundKind exprKind = expr.KindIgnoringSuppressions();
```

Comment:
This approach of sometimes caring about the suppression and sometimes not feels brittle. I'm looking through the code wondering how many uses of `BoundNode.Kind` are now wrong. Did we consider other approaches like making suppression a flag, altering the type, etc ...?